### ISSUE TITLE
[KIEKER-728] Update the Userguide

### ISSUE THREAD

**Issue by rju** [kieker-monitoring/kieker#739]
JIRA Issue: https://kieker-monitoring.atlassian.net/browse/KIEKER-728
Original Reporter: Jan Waller (JIRA username)
 *** 
<p>Update the userguide with the new changes.</p>

### 557058:78303123-c689-4415-9945-bb0c79591440 (JIRA username) added a comment on Tue, 16 Oct 2012 16:30:00
 <p>nie:<br/>
&gt; I finished my review of the examples(&#33;) within the userguide under Windows. I will do the UNIX part later, as I don't have a running UNIX currently available. My review bases on the nightly build 829. What caught my eye:<br/>
&gt; * The traceAnalysis tool delivers a traceAnalysis.kax, but it is neither mentioned nor listed in chapter 5.2. <br/>
&gt; * I wasn't able to use the pic2plot tool to convert to pdf, but I assume this is a specific problem with my own installation.</p>

 ### 557058:629db7ce-5cef-419f-82f8-3dcdebf76471 (JIRA username) added a comment on Tue, 19 Feb 2013 14:40:36
 <p>Update the new API</p>

 ### 557058:629db7ce-5cef-419f-82f8-3dcdebf76471 (JIRA username) added a comment on Mon, 25 Feb 2013 17:08:47
 <p>Chapter 4 has to be updated completely. The new constructors are used, but the old usage (with the register-methods) is still mentioned. Together with incorrect line numbers. That means:</p>
<ul>
	<li>Figure 4.2 has to be modified. It shows the <tt>AnalysisController</tt> and the <tt>register</tt> methods. Use <tt>IProjectContext</tt> and the new constructors.</li>
	<li>The mentioned steps are now wrong.</li>
	<li>It has to be mentioned that the components are registered by using the constructor <br class="atl-forced-newline" />- and that it is not possible to use a components within two different controllers.</li>
	<li>Update Figure 4.3</li>
	<li>We have to check the lines of all listings.</li>
</ul>

 ### 557058:629db7ce-5cef-419f-82f8-3dcdebf76471 (JIRA username) added a comment on Mon, 25 Feb 2013 17:10:35
 <p>Take also a look at</p>
<ul>
	<li>
    <span class="jira-issue-macro resolved" data-jira-key="KIEKER-800" >
                <a href="https://kieker-monitoring.atlassian.net/browse/KIEKER-800" class="jira-issue-macro-key issue-link"  title="Add documentation for colored traces, tooltips, response times" >
            <img class="icon" src="https://kieker-monitoring.atlassian.net/images/icons/issuetypes/genericissue.png" />
            KIEKER-800
        </a>
                                                    <span class="aui-lozenge aui-lozenge-subtle aui-lozenge-success jira-macro-single-issue-export-pdf">Done</span>
            </span>
 (Documentation for the colored traces)</li>
	<li>
    <span class="jira-issue-macro" data-jira-key="KIEKER-977" >
                <a href="https://kieker-monitoring.atlassian.net/browse/KIEKER-977" class="jira-issue-macro-key issue-link"  title="Restructure user guide" >
            <img class="icon" src="https://kieker-monitoring.atlassian.net/rest/api/2/universal_avatar/view/type/issuetype/avatar/10303?size=medium" />
            KIEKER-977
        </a>
                                                    <span class="aui-lozenge aui-lozenge-subtle aui-lozenge-complete jira-macro-single-issue-export-pdf">Open</span>
            </span>
 (Restructuring of the user guide) including 
    <span class="jira-issue-macro resolved" data-jira-key="KIEKER-699" >
                <a href="https://kieker-monitoring.atlassian.net/browse/KIEKER-699" class="jira-issue-macro-key issue-link"  title="Kieker Feedback from SEPVS Lecture (German)" >
            <img class="icon" src="https://kieker-monitoring.atlassian.net/images/icons/issuetypes/genericissue.png" />
            KIEKER-699
        </a>
                                                    <span class="aui-lozenge aui-lozenge-subtle aui-lozenge-success jira-macro-single-issue-export-pdf">Done</span>
            </span>
 (Feedback from the SEPVS lecture)</li>
</ul>

 ### 557058:9e328f65-6e51-4aca-812b-82e63a91a0ae (JIRA username) added a comment on Tue, 5 Mar 2013 11:17:56
 <p>@nie: Could you take care of the changes with respect to the API changes, please? (comment:3)</p>

 ### 557058:629db7ce-5cef-419f-82f8-3dcdebf76471 (JIRA username) added a comment on Sat, 9 Mar 2013 10:54:53
 <p>I updated the line numbers and made sure that the guide refers to the constructor with both arguments (instead to the register methods). I also modified the mentioned steps for the analysis. Some images have still to be updated. </p>

<p>Anything else missing (Except for the fact that we have to restructure the user guide anyway some day) with respect to the API changes?</p>

 ### 557058:9e328f65-6e51-4aca-812b-82e63a91a0ae (JIRA username) added a comment on Sat, 9 Mar 2013 15:10:37
 <p>Replying to <span class="error">&#91;nie|comment:6&#93;</span>:<br/>
&gt; I updated the line numbers and made sure that the guide refers to the constructor with both arguments (instead to the register methods). I also modified the mentioned steps for the analysis. Some images have still to be updated.</p>

<p>thanks. The images you are referring to are those related to the Analysis API changes?</p>

<p>&gt; <br/>
&gt; Anything else missing (Except for the fact that we have to restructure the user guide anyway some day) with respect to the API changes?</p>

<p>Björn will prepare a draft for documenting the extension to the adaptive monitoring feature (
    <span class="jira-issue-macro resolved" data-jira-key="KIEKER-485" >
                <a href="https://kieker-monitoring.atlassian.net/browse/KIEKER-485" class="jira-issue-macro-key issue-link"  title="Adaptive monitoring: Extend set of allowed patterns" >
            <img class="icon" src="https://kieker-monitoring.atlassian.net/images/icons/issuetypes/genericissue.png" />
            KIEKER-485
        </a>
                                                    <span class="aui-lozenge aui-lozenge-subtle aui-lozenge-success jira-macro-single-issue-export-pdf">Done</span>
            </span>
)</p>

 ### 557058:629db7ce-5cef-419f-82f8-3dcdebf76471 (JIRA username) added a comment on Sat, 9 Mar 2013 16:02:42
 <p>Replying to <span class="error">&#91;avh|comment:7&#93;</span>:<br/>
&gt; thanks. The images you are referring to are those related to the Analysis API changes?</p>

<p>Yes, indeed. Shouldn't be much though:</p>
<ul>
	<li>Figure 1.2 shows the AnalysisController. Maybe we should refer to the IAnalysisController (This decision would affect other images as well)?</li>
	<li>Figure 4.2 shows the old methods and constructors.</li>
	<li>Figure 4.3 shows the old constructor.</li>
</ul>

 ### 557058:629db7ce-5cef-419f-82f8-3dcdebf76471 (JIRA username) added a comment on Sun, 17 Mar 2013 17:27:46
 <p>Technically I think Figure 1.1 has to be changed as well. Our analysis controller supports now more than one reader.</p>

<p>Edit: Fig 1.3. It looks like there is one blank too much in "&lt;your interception technology&gt;". It looks also like "Resource utilization" is incorrectly intended.</p>

 ### 557058:629db7ce-5cef-419f-82f8-3dcdebf76471 (JIRA username) added a comment on Mon, 18 Mar 2013 10:57:57
 <p>Fig 3.4 has to be modified as well. I think we have some new writers (zip-writer etc.)</p>

 ### 557058:629db7ce-5cef-419f-82f8-3dcdebf76471 (JIRA username) added a comment on Mon, 18 Mar 2013 11:28:20
 <p>From the current version of the userguide:</p>

<div class="preformatted panel" style="border-width: 1px;"><div class="preformattedContent panelContent">
<pre>As soon as the reader returns from the execution of its read method,
the method terminate of each registered plugin is called by the AnalysisController
</pre>
</div></div>

<p>The AC can have more than one reader currently. How is this handled at the moment, if one of them terminates?</p>

 ### 557058:629db7ce-5cef-419f-82f8-3dcdebf76471 (JIRA username) added a comment on Mon, 18 Mar 2013 11:30:27
 <p>Figure 4.3 doesn't show the new FSZipReader.</p>

 ### 557058:78303123-c689-4415-9945-bb0c79591440 (JIRA username) added a comment on Mon, 18 Mar 2013 13:11:40
 <p>Replying to <span class="error">&#91;nie|comment:12&#93;</span>:<br/>
&gt; From the current version of the userguide:<br/>
&gt; <br/>
&gt; <br/>
&gt; {{</p>
{
&gt; As soon as the reader returns from the execution of its read method,
&gt; the method terminate of each registered plugin is called by the AnalysisController
&gt; }
<p>}}<br/>
&gt; <br/>
&gt; The AC can have more than one reader currently. How is this handled at the moment, if one of them terminates?</p>

<p>Each reader is started in an own thread. The <tt>AC</tt> waits on a latch till all threads are done. On errors all other readers are notified to stop.</p>

 ### 557058:629db7ce-5cef-419f-82f8-3dcdebf76471 (JIRA username) added a comment on Wed, 20 Mar 2013 16:30:48
 <p>Could anyone take a look at Figure 4.2? There are still the extension arrows between the interfaces missing and the picture is already way too confusing. Any idea how to simplify this?</p>

 ### 557058:9e328f65-6e51-4aca-812b-82e63a91a0ae (JIRA username) added a comment on Wed, 3 Apr 2013 14:31:58
 <p>Replying to <span class="error">&#91;nie|comment:16&#93;</span>:<br/>
&gt; Could anyone take a look at Figure 4.2? There are still the extension arrows between the interfaces missing and the picture is already way too confusing. Any idea how to simplify this?</p>

<p>We should at least decrease the width (and increase the height) of the figure, e.g., by positioning <tt>AbstractReaderPlugin</tt> and <tt>AbstractRepository</tt> partly below <tt>AbstractFilterPlugin</tt>. Currently, the fonts are way too small.</p>

 ### 557058:629db7ce-5cef-419f-82f8-3dcdebf76471 (JIRA username) added a comment on Thu, 4 Apr 2013 16:13:40
 <p>With b0390e252d997a117dba4fa3fa14cca59f30c05d I updated the remaining images. The part about the adaptive monitoring is still missing (bjw should write this currently).</p>

 ### 557058:629db7ce-5cef-419f-82f8-3dcdebf76471 (JIRA username) added a comment on Fri, 5 Apr 2013 17:38:26
 <p>bjw added the part about the adaptive monitoring in fb7c7c4dce16f4c8e2eff4e1608f219d01890083; I also modified the formatting in bcaebb1a26d5c2fafa16be503476ce6ab29b8781.</p>

 ### 557058:9e328f65-6e51-4aca-812b-82e63a91a0ae (JIRA username) added a comment on Wed, 10 Apr 2013 14:20:10
 <p>Many, many thanks, Nils&#33; </p>

<p>Minor fixes added in changeset:b08cba1bb7c78364ce6307cce9c616a347cd1e1b/kieker-git <em>include</em>:</p>

<ul>
	<li>Titlepage:</li>
	<li>Added University of Stuttgart</li>
	<li>Removed CAU logo</li>
</ul>

<p>The question I see here is whether to include this information at all and whether to include the university logos. Removing the University information would make the titlepage pretty empty. Adding logos looks nice but the main logo should be the Kieker logo</p>

<ul>
	<li>Added Section 1.5 "Kieker is recommended by the SPEC Research Group"</li>
	<li>(Now) 1.6: Added examples for Appendix contents (Java EE, JMS, Sigar)</li>
	<li>Switched 3.1.6 (before: JMX) and 3.1.7 (before: Adaptive); ref to (now) 3.1.6 in intro of 3.1</li>
</ul>

 ### 557058:629db7ce-5cef-419f-82f8-3dcdebf76471 (JIRA username) added a comment on Wed, 10 Apr 2013 14:37:24
 <p>Take a look at 
    <span class="jira-issue-macro resolved" data-jira-key="KIEKER-906" >
                <a href="https://kieker-monitoring.atlassian.net/browse/KIEKER-906" class="jira-issue-macro-key issue-link"  title="Binay release: Already include libs required for examples in examples&#39; `lib/` dirs" >
            <img class="icon" src="https://kieker-monitoring.atlassian.net/rest/api/2/universal_avatar/view/type/issuetype/avatar/10303?size=medium" />
            KIEKER-906
        </a>
                                                    <span class="aui-lozenge aui-lozenge-subtle aui-lozenge-success jira-macro-single-issue-export-pdf">Done</span>
            </span>
.</p>

 ### 557058:78303123-c689-4415-9945-bb0c79591440 (JIRA username) added a comment on Wed, 10 Apr 2013 14:44:17
 <p>Replying to <span class="error">&#91;avh|comment:23&#93;</span>:<br/>
&gt;   * Removed CAU logo</p>

<p>So we should also remove the file if no longer needed?</p>

 ### 557058:9e328f65-6e51-4aca-812b-82e63a91a0ae (JIRA username) added a comment on Wed, 10 Apr 2013 14:47:45
 <p>Replying to <span class="error">&#91;jwa|comment:25&#93;</span>:<br/>
&gt; Replying to <span class="error">&#91;avh|comment:23&#93;</span>:<br/>
&gt; &gt;   * Removed CAU logo<br/>
&gt; <br/>
&gt; So we should also remove the file if no longer needed?</p>

<p>Done (changeset:bc3939329138f3a7b18d79cc843b16c57e099c7b/kieker-git)</p>
